src/lib/utils/options/declaration.ts

```diff
@@ -72,29 +73,36 @@ export class OptionDeclaration {
         return result;
     }
 
-    convert(value: any, errorCallback?: Function): any {
+    /**
+     *
+     * @param value the value the user passed in
+     * @param errorCallback
+     */
+    convert(value: unknown, errorCallback?: (format: string, ...args: string[]) => void): any {
         switch (this.type) {
             case ParameterType.Number:
-                value = parseInt(value, 10);
+                value = parseInt(value + '', 10);
                 break;
             case ParameterType.Boolean:
-                value = (typeof value === void 0 ? true : !!value);
+                value = !!value;
```

Comment:
`typeof value` will *always* return a string, so it will never be strictly equal to `undefined`. This was dead code.

Reply:
Ah, that makes sense. 👍

Reply:
I don't see how `typeof value` will always return a string. I'm also wondering about if empty strings should be set to true. That type of change might be better in s a separate PR. If there is no error, I'd prefer to leave this change out of this PR.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

Reply:
I agree that this should be reviewed as it was likely a mistake when first added, but this change does not modify existing behavior. In fact, tslint's [`typeof-compare`](https://palantir.github.io/tslint/rules/typeof-compare/) rule should probably be enabled as it will catch this type of mistake in the future.

```js
for (const value of [ 's', 1, {}, true, null, undefined ]) {
  console.log(typeof value, typeof value === void 0)
}
```

Reply:
Opened #885 to follow up on this discussion.